mllib/src/main/scala/org/apache/spark/ml/tree/impl/RandomForest.scala

```diff
@@ -931,7 +925,8 @@ private[spark] object RandomForest extends Logging {
     val numFeatures = metadata.numFeatures
     val splits: Array[Array[Split]] = Array.tabulate(numFeatures) {
       case i if metadata.isContinuous(i) =>
-        val split = continuousSplits(i)
+        // some features may only contains zero, so continuousSplits will not have a record
+        val split = if (continuousSplits.contains(i)) continuousSplits(i) else Array.empty[Split]
```

Comment:
Just `continuousSplits.getOrElse(i, Array.empty[Split])`? Not that it matters, but also avoids searching the map twice.

Reply:
have fixed it